flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/failover/FailoverRegion.java

```diff
@@ -206,13 +227,21 @@ private void restart(long globalModVersionOfFailover) {
 		try {
 			if (transitionState(JobStatus.CREATED, JobStatus.RUNNING)) {
 				// if we have checkpointed state, reload it into the executions
-				//TODO: checkpoint support restore part ExecutionVertex cp
-				/**
 				if (executionGraph.getCheckpointCoordinator() != null) {
+					// we restart the checkpoint scheduler for
```

Comment:
As discussed offline, instead of doing this in the `FailoverRegion` wouldn't it be better to just have a method like `CheckpointCoordinator::cancelPendingCheckpoints` that only iterates and cancels the pending checkpoints and clears the pending checkpoint map while holding the `lock`?

Reply:
In the future, we probably also want to have this clearing of pending checkpoints in general before applying any failover strategy because the problem can also exist for more strategies.

Reply:
I believe we would fix this when supporting individual failover.